main/src/ca/uwaterloo/flix/language/phase/unification/Substitution.scala

```diff
@@ -30,12 +30,12 @@ object Substitution {
   /**
     * Returns the singleton substitution mapping the type variable `x` to `tpe`.
     */
-  def singleton(x: Type.KindedVar, tpe: Type): Substitution = {
+  def singleton(x: Symbol.TypeVarSym, tpe: Type): Substitution = {
```

Comment:
Whats the advantanges / disadvantages of using the symbol directly here instead of using Type.KindedVar?

Reply:
The main reason for this choice is that semantically it makes sense. A `TypeVarSym` represents the declaration/identity of a type variable, while a `KindedVar` represents a usage of it. When we substitute, we look for any reference to a variable, not just a reference to one particular usage of it.

It also follows closely from using TypeVarSyms in type parameters and schemes.
I convinced myself with the analogy:

`Expression.Var R Symbol.VarSym`
`Type.KindedVar R Symbol.KindedTypeVarSym`

for some relation `R` :)